cordova-plugin-outline/apple/vpn/PacketTunnelProvider.m

```diff
@@ -117,55 +114,53 @@ - (void)startTunnelWithOptions:(NSDictionary *)options
-                                                                                       : errorCode;
-                         if (clientErrorCode == noError) {
-                           [self connectTunnel:[self getTunnelNetworkSettings]
-                                    completion:^(NSError *error) {
-                                      if (!error) {
-                                        BOOL isUdpSupported =
-                                            isOnDemand ? self.connectionStore.isUdpSupported
-                                                       : errorCode == noError;
-
-                                        if (![self startTun2Socks:isUdpSupported]) {
-                                          [self execAppCallbackForAction:kActionStart
-                                                               errorCode:vpnStartFailure];
-                                          return completionHandler([NSError
-                                              errorWithDomain:NEVPNErrorDomain
-                                                         code:NEVPNErrorConnectionFailed
-                                                     userInfo:nil]);
-                                        }
-                                        [self execAppCallbackForAction:kActionStart
-                                                             errorCode:noError];
-                                        [self listenForNetworkChanges];
-                                        [self.connectionStore save:connection];
-                                        self.connectionStore.isUdpSupported = isUdpSupported;
-                                        self.connectionStore.status = ConnectionStatusConnected;
-
-                                      } else {
-                                        [self execAppCallbackForAction:kActionStart
-                                                             errorCode:vpnPermissionNotGranted];
-                                      }
-                                      completionHandler(error);
-                                    }];
-                         } else {
-                           [self execAppCallbackForAction:kActionStart errorCode:clientErrorCode];
-                           completionHandler([NSError errorWithDomain:NEVPNErrorDomain
-                                                                 code:NEVPNErrorConnectionFailed
-                                                             userInfo:nil]);
-                         }
-                       }];
+  long errorCode = noError;
+  __block NSError *err = nil;
+  if (!isOnDemand) {
+    ShadowsocksCheckConnectivity(self.hostNetworkAddress, [self.connection.port intValue],
+                                 self.connection.password, self.connection.method, &errorCode,
+                                 &err);
+  }
+  __block ErrorCode clientErrorCode =
```

Comment:
I think the logic might be clearer without this temporary variable.  Instead, the condition would be
```objc
if (errorCode == noError || errorCode == udpRelayNotEnabled) {
```
You might still need `clientErrorCode` to be defined inside the block, but I think that would be more readable than defining it here.

Reply:
Done. There is no need for `clientErrorCode` now.